androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/ManageDataLauncherActivity.java

```diff
@@ -332,25 +334,65 @@ public static boolean packageSupportsSiteSettings(String packageName,
      *
      * Otherwise returns null if {@link ManageDataLauncherActivity} is not launchable
      * or if shortcuts are not supported by the Android SDK version.
+     *
+     * The shortcut returned does not specify an activity. Thus when the shortcut is added,
+     * the app's main activity will be used by default. This activity needs to define the
+     * MAIN action and LAUNCHER category in order to attach the shortcut.
      */
-    public static ShortcutInfo getSiteSettingsShortcutOrNull(Context context, PackageManager packageManager) {
+    @Nullable
+    public static ShortcutInfo getSiteSettingsShortcutOrNull(Context context,
+            PackageManager packageManager) {
         Intent siteSettingsIntent = new Intent(context, ManageDataLauncherActivity.class);
         siteSettingsIntent.setAction(ACTION_MANAGE_TRUSTED_WEB_ACTIVITY_DATA);
         List<ResolveInfo> activities = packageManager.queryIntentActivities(siteSettingsIntent,
                 PackageManager.MATCH_DEFAULT_ONLY);
-        if(activities.size() > 0) {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
-                return new ShortcutInfo.Builder(context, ACTION_MANAGE_TRUSTED_WEB_ACTIVITY_DATA)
-                        .setShortLabel("Site Settings")
-                        .setLongLabel("Manage website notifications, permissions, etc.")
-                        .setIcon(Icon.createWithResource(context,
-                                android.R.drawable.ic_menu_preferences))
-                        .setIntent(siteSettingsIntent)
-                        .build();
-            }
+
+        if(activities.size() == 0) return null;
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
```

Comment:
We can invert this check and then move it to the top of the method - there's no point querying the package manager if we know we can't create a shortcut.